src/Polysemy/Internal/Union.hs

```diff
@@ -265,14 +265,13 @@ decomp (Union p a) =
 -- | Retrieve the last effect in a 'Union'.
 extract :: Union '[e] m a -> Weaving e m a
 extract (Union Here a)   = a
-extract (Union (There g) _) = case g of {}
 {-# INLINE extract #-}
 
 
 ------------------------------------------------------------------------------
 -- | An empty union contains nothing, so this function is uncallable.
 absurdU :: Union '[] m a -> b
-absurdU (Union pr _) = case pr of {}
```

Comment:
Is making this less strict an issue?

Reply:
Doesn't GHC show a warning about non-exhaustive match?

Reply:
It seems `ghc` can now tell that `ElemOf` (and by extension `Union`) cannot produce a value with an empty list as its index! Very nice to see this

Reply:
Ah, I see - really nice! But I guess that it may be more convenient to just keep it as it was while we're supporting older versions, because it doesn't really change anything interesting about the behaviour.

Reply:
To be honest I would also prefer to have `-Werror` in CI to catch other mistakes, but then we would definitely require `ifdef`s here.

Reply:
I mean, the older version doesn't emit any warnings on new GHC, does it?